colossalai/inference/core/request_handler.py

```diff
@@ -159,40 +187,34 @@ def schedule(self):
                             remove_list.append(seq)
                             break
 
-                        # stop feeding new sequence into running list to assure
-                        if self.cache_manager.num_available_blocks <= self.running_list.total_seq_num():
-                            break
+                    num_seqs_to_add = min(len(lst), self.max_batch_size - self.running_list.total_seq_num())
+                    remove_list.extend(lst[:num_seqs_to_add])
+                    self.running_list.extend(lst[:num_seqs_to_add])
 
-                        # Try to allocate cache blocks for the sequence.
-                        if (
-                            self.cache_manager.check_allocation(seq)
-                            and (len(self.running_list.prefill) + len(self.running_list.decoding))
-                            < self.max_batch_size  # There some bugs in continous batching, so we disable it here.
-                        ):
-                            # If succeed, add the sequence to running list.
-                            remove_list.append(seq)
-                            self.running_list.append(seq)
-                            self.cache_manager.allocate_context_from_block_table(seq.block_table, seq.sentence_len)
                     for seq in remove_list:
                         lst.remove(seq)
 
         if self.running_list.ready_for_prefill():
-            for seq in self.running_list.prefill:
-                seq.mark_running()
-            self.prefill_batch.add_seqs(self.running_list.prefill)
-            return self.prefill_batch
-
-        if not self.running_batch.is_empty:
-            for seq in self.running_batch.sequences_set:
-                recycle = self.cache_manager.allocate_token_from_block_table(seq.block_table, seq.sentence_len)
-                if recycle:
+            self.running_list.mark_prefill_running()
+            # allocate blocks for the prefill batch
+            self.prefill_bb.add_seqs(
+                self.running_list.prefill, alloc_block_tables_fn=self.cache_manager.allocate_context_from_block_tables
```

Comment:
some questions here, if the profill list is larger than `max_batch_size`, some seas will be reserved. they should not be marked running, maybe try to move the mark function to a new place

Reply:
Revised by adding available seqs while keeping seqs which the batch bucket cannot hold in the pool (RunningList). When moving prefill to decoding, we'll only move those specific seqs, instead of moving all the sequences in RunningList.prefill.

Reply:
lgtm